### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by nikhilMHI**
Hi @sherriously

Have cleaned up the branches and formatted code using prettier

Thanks,
Nikhil

### CODE REVIEW

File: mhi_bg_customglplugin_final.js [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+      "Processing stTranType = " + stTranType + ", stRecordId = " + stRecordId
+    );
+
+    var createdfrom = transactionRecord.getFieldValue("createdfrom");
+    var createdfromtext = transactionRecord
+      .getFieldText("createdfrom")
+      .substr(0, 11);
+    var ifstatus = transactionRecord.getFieldValue("shipstatus");
+
+    if (createdfromtext != "Sales Order" || ifstatus != "C") {
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: store status `C` in a constant e.g. `var SHIPPED = 'C';`

File: mhi_bg_customglplugin_final.js [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+      nlapiLogExecution(
+        "DEBUG",
+        stLogTitle,
+        "Exiting as IF is not created from sales order or status is not shipped"
+      );
+      return;
+    }
+
+    var rec = nlapiLoadRecord("salesorder", createdfrom);
+    // var bOrderType = nlapiLookupField('salesorder',createdfrom, 'custbody_bg_ordertype');
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: avoid leaving commented code in the script unless this is something you might turn on/off in the future

File: mhi_bg_customglplugin_final.js [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+    var bOrderType = rec.getFieldValue("custbody_bg_ordertype");
+    //
+    //var bProject = nlapiLookupField('salesorder',createdfrom, 'job');
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: same here

File: mhi_bg_customglplugin_final.js [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+
+    if (isEmpty(orderTypeExpenseAcc)) {
+      nlapiLogExecution("DEBUG", stLogTitle, "no order type found so exit");
+      return;
+    }
+
+    var objItem = getCogsAccount(transactionRecord);
+    nlapiLogExecution("DEBUG", stLogTitle, objItem);
+
+    //   assessTransactionLines(transactionRecord, standardLines, customLines, book);
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: same here

File: mhi_bg_customglplugin_final.js [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+        // figure out the actualcogs account to be used based on stEntityExpenseAcc parameters
+        var ordertype = [];
+        var actualcogsaccount = [];
+        var item = [];
+        var customrecordid = [];
+        var cogs = [];
+
+        var ordertypedata = {};
+
+        //nlapiLogExecution('DEBUG', 'length of array for ordertype custom record', stEntityExpenseAcc.length);
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove

File: mhi_bg_customglplugin_final.js [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+        for (var a = 0; a < stEntityExpenseAcc.length; a++) {
+          ordertypedata.item = stEntityExpenseAcc[a].item;
+          ordertypedata.project = stEntityExpenseAcc[a].project;
+          ordertypedata.ordertype = stEntityExpenseAcc[a].ordertype;
+          ordertypedata.actualcogsaccount =
+            stEntityExpenseAcc[a].actualcogsaccount;
+
+          //nlapiLogExecution('DEBUG', stLogTitle, 'cogs account value' + stEntityExpenseAcc[a].actualcogsaccount);
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove

File: mhi_bg_customglplugin_final.js [hunk 7]
```diff
@@ -0,0 +1,481 @@
...
+
+          ordertypedata.cogs = stEntityExpenseAcc[a].actualcogsaccount;
+          ordertypedata.customrecordid = stEntityExpenseAcc[a].customrecordid;
+
+          //if((stEntityExpenseAcc[a].item == stItem) && (stEntityExpenseAcc[a].ordertype == bOrderType))
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove

File: mhi_bg_customglplugin_final.js [hunk 8]
```diff
@@ -0,0 +1,481 @@
...
+ *
+ * This software is the confidential and proprietary information of
+ * MyersHolum, Inc. ('Confidential Information'). You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with MyersHolum.
+ *
+ * Custom GL for assessment processing on Item Fulfillment
+ *
+ * Version      Date            Author              Remarks
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: no need for this version control here - use Github instead. Every commit message should just be your "remarks" here

File: mhi_bg_customglplugin_final.js [hunk 9]
```diff
@@ -0,0 +1,481 @@
...
+
+  var objItem = {};
+  for (var i = 1; i <= transactionRecord.getLineItemCount("item"); i++) {
+    var stItem = transactionRecord.getLineItemValue("item", "item", i);
+    var stItemType = transactionRecord.getLineItemValue("item", "itemtype", i);
+
+    nlapiLogExecution("DEBUG", stLogTitle, stItemType);
+    var recordtype = "";
+
+    switch (
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: Encapsulate this in a separate function to determine recordtype

File: mhi_bg_customglplugin_final.js [hunk 10]
```diff
@@ -0,0 +1,481 @@
...
+
+      case "GiftCert":
+        recordtype = "giftcertificateitem";
+        break;
+      default:
+    }
+
+    var stCogsAccnt = nlapiLookupField(recordtype, stItem, "expenseaccount");
+    nlapiLogExecution("DEBUG", stLogTitle, stCogsAccnt);
+    /*
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove

File: mhi_bg_customglplugin_final.js [hunk 11]
```diff
@@ -0,0 +1,481 @@
...
+    null,
+    arrColumns
+  );
+
+  var resultSet = searchAssessmentCharges.runSearch().getResults(0, 1);
+  var stAssSearchId = resultSet[0].getValue(
+    "custrecord_bg_item_order_type_search"
+  );
+  var stAssSearchType = resultSet[0].getValue("custrecord_bg_record_type");
+  /*
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove

File: mhi_bg_customglplugin_final.js [hunk 12]
```diff
@@ -0,0 +1,481 @@
...
+        return false; //Only get the first record
+    });
+*/
+  if (isEmpty(searchAssessmentCharges)) throw "No Parameter Specified";
+
+  var arrAssCodes = [];
+  var objItemCode = {};
+  for (var j = 1; j <= transactionRecord.getLineItemCount("item"); j++) {
+    var stAssCode = transactionRecord.getLineItemValue("item", "item", j);
+    //	var stItem = transactionRecord.getLineItemValue('item', 'item', j);
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove commented out code

File: mhi_bg_customglplugin_final.js [hunk 13]
```diff
@@ -0,0 +1,481 @@
...
+  return ordertypecogsacc;
+}
+
+function getAssessmentCharges(object) {
+  var stLogTitle = "getAssessmentCharges";
+
+  // load order type-cogs account record for the item and get the related data
+  var objSearch = nlapiLoadSearch(object.stAssSearchType, object.stAssSearchId);
+  var arrFilter = [];
+  // arrFilter.push(new nlobjSearchFilter('custrecord_bg_item',null,'anyof',object.stAssessmentCode));
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove

File: mhi_bg_customglplugin_final.js [hunk 14]
```diff
@@ -0,0 +1,481 @@
...
+      "custrecord_bg_cogs_account"
+    );
+    objAssessmentChargeData.item = arrResults[i].getValue("custrecord_bg_item");
+    objAssessmentChargeData.customrecordId = arrResults[i].getValue(
+      "internalid"
+    );
+    objAssessmentChargeData.project = arrResults[i].getValue(
+      "custrecord_bg_project"
+    );
+    //  objAssessmentChargeData.cogs = arrResults[i].getValue("expenseaccount","custrecord_bg_item",null);
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: remove

**Comment by nikhilMHI** [comment]
> m not sure what you are using to upload files to Github but there should be a detailed commit message

Hi Sherri,

Will review and clean it up as some of the commented code may be needed later . Let me go through each comment and resolve it based on what needs to happen.

Thanks,
Nikhil

**Comment by nikhilMHI** [comment]
Resolved all the conversations

**Comment by nikhilMHI** [comment]
Hi @sherriously have cleaned it up as per comments and also mentioned my comments under the latest commit

Thanks for your time to review

Nikhil

File: mhi_bg_customglplugin_final.js [hunk 15]
```diff
@@ -37,8 +36,9 @@ function customizeGlImpact(
       .getFieldText("createdfrom")
       .substr(0, 11);
     var ifstatus = transactionRecord.getFieldValue("shipstatus");
+    var Shipped = "C";
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: use all caps for constants, e.g. `var SHIPPED = 'C';`

File: mhi_bg_customglplugin_final.js [hunk 16]
```diff
@@ -48,27 +48,24 @@ function customizeGlImpact(
...
 
     var rec = nlapiLoadRecord("salesorder", createdfrom);
-    // var bOrderType = nlapiLookupField('salesorder',createdfrom, 'custbody_bg_ordertype');
-    var bOrderType = rec.getFieldValue("custbody_bg_ordertype");
-    //
-    //var bProject = nlapiLookupField('salesorder',createdfrom, 'job');
-    var bProject = rec.getFieldValue("job");
 
-    if (isEmpty(bOrderType) || isEmpty(bProject)) {
+    var OrderType = rec.getFieldValue("custbody_bg_ordertype");
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: variable names should start lowercase `orderType` is great

File: mhi_bg_customglplugin_final.js [hunk 17]
```diff
@@ -48,27 +48,24 @@ function customizeGlImpact(
...
+
+    var Project = rec.getFieldValue("job");
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: `var project`

File: mhi_bg_customglplugin_final.js [hunk 18]
```diff
@@ -424,14 +358,41 @@ function getAssessmentCharges(object) {
...
     );
-    //  objAssessmentChargeData.cogs = arrResults[i].getValue("expenseaccount","custrecord_bg_item",null);
 
     arrAssessmentLines.push(objAssessmentChargeData);
   }
 
   return arrAssessmentLines;
 }
 
+function getItemType(stItemType) {
```
Comment (sherriously) [inline mhi_bg_customglplugin_final.js]: amazing

**Comment by nikhilMHI** [comment]
Hi @sherriously, 

have performed the updates requested

Thanks,
Nikhil

**Pull request merged**